src/cabal-details.js

```diff
@@ -408,6 +412,66 @@ class CabalDetails extends EventEmitter {
     }
   }
 
+  /**
+   * Archive a channel. Publishes a message announcing
+   * that you have archived the channel, applying it to the views of others who have you as a moderator/admin.
+   * @param {string} channel
+   * @param {string} [reason]
+   * @param {function} cb - callback invoked when the operation has finished, with error as its only parameter
+   */
+  archive (channel, reason = "", cb) {
+    if (!cb) cb = noop
+    const details = this.channels[channel]
+
+    if (channel === '!status') {
+      return nextTick(cb, new Error('cannot archive the !status channel'))
+    }
+    if (!details) {
+      return nextTick(cb, new Error('cannot archive non-existent channel'))
+    }
+    this.channels[channel].archive()
+    this.publishMessage({
+      type: 'channel/archive',
+      content: {
+        channel,
+        reason
+      }
+    }, {}, (err) => {
+      if (err) cb(err)
```

Comment:
is there a reason you don't want to pass the falsey error through to the callback? i would expect the arg to be `undefined` if it was falsey, which would make it equivalent in either case.

this would also have the same issue in the case where the error is swallowed when the callback isn't supplied - maybe throw some logging in here too.

Reply:
good point, fixed (tho w/o logging—feel free to clarify why you think it would be important? not following atm)

Reply:
logging isn't necessary! i just meant in the case where there is a non-obvious situation where `err` is falsey you should log out that state (or at least add a comment).

marking as resolved.

Reply:
ahhhh i understand now, thanks! that is a good idea yeah